### PR TITLE
grpc for pod remove

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: Yao Zengzeng yaozengzeng@zju.edu.cn

### CODE REVIEW

**Comment by resouer** [comment]
@YaoZengzeng  You don't need to add a separate file `pkg/kubelet/hyper/types/types.proto`, you can import it as dependency, like: `github.com/hyperhq/hyperd/types`

**Comment by feiskyer** [comment]
@resouer Import `github.com/hyperhq/hyperd/types` as a dependency requires hypernetes and hyperd are based completely same version of grpc and its dependencies. It's better to copy the proto file and generate `.pb.go` here to avoid this problem.

@YaoZengzeng It's better to add a script to generate and verify hyperd api.

**Comment by feiskyer** [comment]
Sorry for wrongly closing this, reopen it.

**Comment by resouer** [comment]
@YaoZengzeng Please update this PR and fix CI

File: pkg/kubelet/hyper/hyperclient.go [hunk 1]
```diff
@@ -483,23 +487,20 @@ func (client *HyperClient) RemoveImage(imageID string) error {
...
-	v := url.Values{}
-	v.Set(KEY_POD_ID, podID)
-	body, _, err := client.call("DELETE", "/pod?"+v.Encode(), "", nil)
-	if err != nil && !strings.Contains(err.Error(), "Can not find") {
-		return err
-	}
-
-	var result podRemoveResult
-	err = json.Unmarshal(body, &result)
+	conn, err := grpc.Dial(HYPER_SERVER, grpc.WithInsecure())
```
Comment (resouer) [inline pkg/kubelet/hyper/hyperclient.go]: Move all connection dial process to `client *HyperClient`, so in these handlers you just exec `cli.PodRemove`

Comment (resouer) [reply pkg/kubelet/hyper/hyperclient.go]: @YaoZengzeng A good example is https://github.com/kubernetes/frakti/blob/master/pkg/hyper/client.go

**Comment by resouer** [comment]
Please check all places referenced `NewHyperClient` since you changed its return value.

**Comment by resouer** [comment]
lgtm

**Pull request merged**